### DISCUSSION TITLE
Can I get the complete sql in a db.statement without '?', or get the data type of db.sql.parameters?

### DISCUSSION THREAD

**Discussion by Silvengs** [apache/skywalking#12697] [Q&A] [answered]
example: 
`
  tags {
    key: "db.statement"
    value: "SELECT id, username, password FROM t_user WHERE id = ?"
  }
  tags {
    key: "db.sql.parameters"
    value: "[1]"
  }
`

`id = ?`，I don't know the data type of id, `value: 1` could be an integer or a string, I can't recover the original sql.

**Comment by wu-sheng** [comment, marked as answer]
No, you can't. That thing doesn't exist. The JDBC prepared statement and database run in this way.

**Reply by wu-sheng** [reply]
This is for tracing, not to build an SQL. This is not an APM feature.

**Reply by Silvengs** [reply]
thank you for response
